packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,437 @@
+    }
+
+    /// @notice Returns the list of all scheduled but not cancelled or executed transactions for
+    /// for a given safe
+    /// @dev WARNING: This operation will copy the entire set of pending transactions to memory,
+    /// which can be quite expensive. This is designed only to be used by view accessors that are
+    /// queried without any gas fees. Developers should keep in mind that this function has an
+    /// unbounded cost, and using it as part of a state-changing function may render the function
+    /// uncallable if the set grows to a point where copying to memory consumes too much gas to fit
+    /// in a block.
+    /// @return List of pending transaction hashes
+    function pendingTransactionsForSafe(Safe _safe) external view returns (ScheduledTransaction[] memory) {
+        SafeState storage safeState = _safeState[_safe];
+
+        bytes32[] memory hashes = safeState.pendingTxHashes.values();
+        ScheduledTransaction[] memory scheduled = new ScheduledTransaction[](hashes.length);
+        for (uint256 i = 0; i < hashes.length; i++) {
+            scheduled[i] = safeState.scheduledTransactions[hashes[i]];
+        }
+        return scheduled;
+    }
+
+    /// @notice Configure the contract as a timelock guard by setting the timelock delay
+    /// @dev This function does not check if the guard is enabled on the Safe. This provides the option to
+    ///      configure the guard prior to enabling it, or to reset the configuration to 0 after disabling.
+    /// @param _timelockDelay The timelock delay in seconds (0 to clear configuration)
+    function configureTimelockGuard(Safe _safe, uint256 _timelockDelay, bytes memory _signatures) external {
+        // Validate timelock delay - must not be longer than 1 year
+        if (_timelockDelay > 365 days) {
+            revert TimelockGuard_InvalidTimelockDelay();
+        }
+
+        // Generate the configuration transaction data
+        bytes memory data = abi.encodeWithSignature("configureTimelockGuard(uint256)", _timelockDelay);
+
+        // Get the nonce of the Safe to be configured
+        uint256 nonce = _safe.nonce();
+        bytes memory configureTxData = _safe.encodeTransactionData(
+            address(_safe), 0, data, Enum.Operation.Call, 0, 0, 0, address(0), address(0), nonce
+        );
+        bytes32 configureTxHash = _safe.getTransactionHash(
+            address(_safe), 0, data, Enum.Operation.Call, 0, 0, 0, address(0), address(0), nonce
+        );
+
+        // Verify signatures using the Safe's signature checking logic
+        // This function call reverts if the signatures are invalid.
+        _safe.checkSignatures(configureTxHash, configureTxData, _signatures);
+
+        // Store the configuration for this safe
+        SafeState storage safeState = _safeState[_safe];
+        safeState.config.timelockDelay = _timelockDelay;
+
+        // Initialize cancellation threshold to 1
+        safeState.cancellationThreshold = 1;
```

Comment:
Not needed if `configureTimelockGuard` is a regular transaction, because `_resetCancellationthreshold` is called in `checkTransaction`.